### DISCUSSION TITLE
Flowdeck v2 Chip select in PX4 stack (linux sbc)

### DISCUSSION THREAD

**Discussion by dirksavage88** [bitcraze/discussions#693] [Q&A] [answered]
Looking at the flowdeck v2 schematic, I see specfic chip selects called out, but none reference the flowdeck. They show CS for sd card and the loco deck.

Is the pmw3901 using chip select? I have a  linux sbc (running px4 stack) that has a spi port but unable to detect pmw3901 despite setting pin_mux for that port to SPI explicitly. I can detect a voltage of around 1.7-1.8 on SCLK, nothing on MISO and MOSI, and 2v on CS. This is showing that CS is not being pulled low and the sensor is not detected? I am really confused on which IO is CS for the flowdeck sensor as it is not called out in the deck compatibility page either: it only shows SCLK, MISO, MOSI...

I'm confused since other pmw3901 boards have chip select called out. Granted these boards are not part of the bitcraze deck ecosystem...I just want to rule out an issue with chip select before blaming PX4's core code. Usually blaming core driver code in PX4 is the last thing to suspect; generally a physical layer issue is #1 root cause.

**Comment by ataffanel** [comment]
On the [flow V2 schematic](https://www.bitcraze.io/documentation/hardware/flow_deck_2/flow-deck-v2-reva.pdf) the chip-select line is named `CS`. It is connected to `IO_3` of the Crazyflie expansion port and to `Pin 19` of the PMW3901.

As for voltages, our decks are designed for using 3.0V for IO/s and the flow deck is no exception. If you want to use lower IO voltage it seems possible but be careful looking at the schematic for how it will affect the VL53L1 chip and what is the drop-out voltage of the 1.8V regulator for the flow sensor.

**Reply by dirksavage88** [reply, marked as answer]
The issue was that VCOM was left floating. I had to connect VCOM to 3.3v to get the optical flow sensor working. The signal name is a bit confusing (especially since there is vcc also) but looking at the schematic it makes sense since VCOM feeds the vreg to supply 1.8v to the flow sensor.
